cpp/src/arrow/engine/substrait/serde.h

```diff
@@ -96,6 +96,26 @@ ARROW_ENGINE_EXPORT Result<std::shared_ptr<acero::ExecPlan>> DeserializePlan(
     const ExtensionIdRegistry* registry = NULLPTR, ExtensionSet* ext_set_out = NULLPTR,
     const ConversionOptions& conversion_options = {});
 
+/// \brief Deserializes a single-relation Substrait Plan message into an execution plan
+///
+/// The output of each top-level Substrait relation will be sent to a caller supplied
+/// consumer function provided by consumer_factory
+///
+/// \param[in,out] plan an ExecNode corresponding to the single toplevel relation in the
+/// Substrait Plan
+/// \param[in] buf a buffer containing the protobuf serialization of a Substrait Plan
+/// message
+/// \param[in] consumer node that consumes the batches produced by each toplevel Substrait
+/// relation
+/// \param[in] registry an extension-id-registry to use, or null for the default one.
+/// \param[out] ext_set_out if non-null, the extension mapping used by the Substrait
+/// \param[in] conversion_options options to control how the conversion is to be done.
+ARROW_ENGINE_EXPORT Status DeserializePlan(
```

Comment:
Why this change

Reply:
Also, shouldn't this return Result object similar to other Deserialize* function?

Reply:
Same rationale as in [this post](https://github.com/apache/arrow/pull/35513#discussion_r1188823725).